src/main/kotlin/br/com/creditas/credit_analysis/controller/ClientController.kt

```diff
@@ -72,4 +73,8 @@ class ClientController(
         return ResponseEntity.ok(clientRepository.findById(id))
     }
 
+    @GetMapping("/find")
+    fun getByCpf(@RequestParam cpf: String): ResponseEntity<Optional<ClientPF>> {
+        return ResponseEntity.ok(clientRepository.findByCpf(cpf))
```

Comment:
tip: you can directly return the object and use annotations for success status code, the code is leaner

Reply:
Is this correct?
    
    @GetMapping("/find")
    @ResponseStatus(HttpStatus.OK)
    fun getByCpf(@RequestParam cpf: String): ClientPF {
        return clientRepository.findByCpf(cpf)
    }